src/lib.rs

```diff
@@ -1454,6 +1556,16 @@ impl ServiceInfo {
         &self.addresses
     }
 
+    /// Insert an additional address to the service
+    pub fn insert_address(&mut self, addr: Ipv4Addr) {
+        self.addresses.insert(addr);
+    }
+
+    /// Remove an address
+    pub fn remove_address(&mut self, addr: &Ipv4Addr) {
```

Comment:
same here, can we name it `remove_ipv4`?

Reply:
sure, will do.